.circleci/config.yml

```diff
@@ -88,15 +92,56 @@ jobs:
           command: |
             . venv/bin/activate
             cd python
-            python3 -m pytest ml4ir/applications/ranking/tests --junitxml=test_results/results.xml
+            COVERAGE_FILE=reports/.coverage.ranking coverage run --include "ml4ir/*" -m pytest ml4ir/applications/ranking/tests/ --junitxml=test_results/results.xml
           no_output_timeout: 45m  # Time for running the tests before we timeout
+      - persist_to_workspace:
+          root: python/reports
+          paths:
+            - .coverage.ranking
       - store_test_results:
           path: python/test_results
 
+  build_test_coverage:
+    docker:
+      - image: circleci/python:3.7.7
+    resource_class: large
```

Comment:
Did you try with a standard Docker container?

I think, we have running costs starting with "large" or the one just below.